examples/02_miscellaneous/plot_1_pearson_correlation.py

```diff
@@ -0,0 +1,326 @@
+    console = Console()
+
+    # Get current transport plan and tensorize features
+    pi = locals["pi"]
+    source_features_tensor = _make_tensor(source_features, device)
+    target_features_tensor = _make_tensor(target_features, device)
+
+    # Compute the Pearson correlation and append it to the list
+    corr = pearson_corr(source_features_tensor, target_features_tensor, pi)
+    corr_bcd_steps.append(corr)
+
+    console.log("Pearson correlation: ", corr)
+
+
+# %%
+# We now fit the mapping using the sinkhorn solver and 5 BCD iterations.
+device = "cpu"
+
+start_time = time.time()
+mapping = FUGW(alpha=0.5, rho=1, eps=1e-4)
+_ = mapping.fit(
+    source_features=source_features_normalized[:n_training_contrasts],
+    target_features=target_features_normalized[:n_training_contrasts],
+    source_geometry=source_geometry_normalized,
+    target_geometry=target_geometry_normalized,
+    source_features_val=source_features_normalized[n_training_contrasts:],
+    target_features_val=target_features_normalized[n_training_contrasts:],
+    init_plan=init_plan_normalized,
+    solver="sinkhorn",
+    solver_params={
+        "nits_bcd": 5,
+    },
+    callback_bcd=partial(
+        correlation_callback,
+        source_features=source_features_normalized[n_training_contrasts:],
+        target_features=target_features_normalized[n_training_contrasts:],
+        device=device,
+    ),
+    verbose=True,
+    device=device,
+)
+total_time = time.time() - start_time
+
+# %%
+# The Pearson correlation relative to each validation contrast and training
+# loss evolution are then plotted for each BCD iteration. Notice how the
+# correlation only improves upon a certain number of iterations, even though
+# the fugw loss keeps decreasing. This is a sign of overfitting, and we
+# might want to stop after fewer iterations.
+
+corr_bcd_steps = np.array(corr_bcd_steps)
+
+mean_corr = np.mean(corr_bcd_steps, axis=1)
+std_corr = np.std(corr_bcd_steps, axis=1)
+
+fig, ax1 = plt.subplots()
+
+color = "tab:red"
+ax1.set_xlabel("BCD Step")
+ax1.set_ylabel("FUGW loss", color=color)
+ax1.plot(mapping.loss_steps, mapping.loss["total"], color=color)
+ax1.tick_params(axis="y", labelcolor=color)
+
+ax2 = ax1.twinx()  # instantiate a second axes that shares the same x-axis
+
+color = "tab:blue"
+ax2.set_ylabel("Pearson correlation", color=color)
+for i in range(len(source_features_normalized[n_training_contrasts:])):
+    ax2.plot(
+        mapping.loss_steps[: len(corr_bcd_steps)],
+        corr_bcd_steps[:, i],
+        color=color,
+        alpha=0.5,
+        linestyle="dashed",
+    )
+ax2.set_label("Pearson correlation")
+
+ax2.plot(
+    mapping.loss_steps[: len(corr_bcd_steps)],
+    mean_corr,
+    color="blue",
+    label="Average correlation",
+)
+ax2.fill_between(
+    mapping.loss_steps[: len(corr_bcd_steps)],
+    mean_corr - std_corr,
+    mean_corr + std_corr,
+    color=color,
+    alpha=0.2,
+    label="1st standard deviation",
+)
+ax2.set_ylim(0, 1)
```

Comment:
1 validation contrast is missing from your plot because it has negative correlation
```suggestion
ax2.set_ylim(-0.5, 1)
```